### DISCUSSION TITLE
Wrong language detection in Zelta repo

### DISCUSSION THREAD

**Discussion by Andrej123456789** [github-linguist/linguist#5804] [Classification] [answered]
[Here is a link of repo](https://github.com/ringwormGO-organization/Zelta)

GitHub says Inno Setup is "bigger" than C# (which has more and bigger files). Also on repo we can found C++ files

**Comment by lildude** [comment, marked as answer]
The breakdown of your repo as Linguist sees it:

```console
$ github-linguist Zelta/ --breakdown
55.67%  8895       Inno Setup
44.33%  7082       C#

Inno Setup:
Zelta-Installer.iss   <--- From above we can see this file alone is 8895 bytes vs 7082 for all your C#

C#:
Zelta/Form1.cs
Zelta/Program.cs
Zelta/Properties/AssemblyInfo.cs
Zelta/Settings.cs
Zelta/obj/Debug/.NETFramework,Version=v4.7.2.AssemblyAttributes.cs

$
```

As you can see, it has only detected the one very large Inno Setup file and your C# files.

Why? Most of your files, like the C++ files you mentioned, are supplied as part of NuGet packages which are third-party packages which Linguist considered as vendored third-party code and thus [ignores](https://github.com/github/linguist/blob/97bc889ce840208652bf09b45f3b7859de43fe8e/lib/linguist/vendor.yml#L318-L319) (most people don't want third-party code counting towards their repo stats).

If you want these to count, you'll need to implement an [override](https://github.com/github/linguist/blob/master/docs/overrides.md) to tell Linguist to not consider these files as vendored third-party code.

The rest of your repo appears to be mostly "data" files because they are either binaries or are not UTF-8 encoded files, or text files that are considered "prose" (eg `.txt`) or "data" (eg `.xml`) which Linguist doesn't count towards the repo stats by default either. Once again, you'll need to implement an override if you want them to be counted.

**Reply by Andrej123456789** [reply]
Thank you for answer, but I set C++ files in .gitattributes but it seems don't work.

**Reply by lildude** [reply]
It wouldn't have any effect as your old override (`*.cpp linguist-language=C++`) only forces the language identification. It won't override Linguist's marking of the files as `vendored`. You would need to override this as all of your C++ files are in the vendored `packages` directory.

**Reply by Andrej123456789** [reply]
Ok

**Reply by whitemike889** [reply]
> The breakdown of your repo as Linguist sees it:
> 
> 
> 
> ```console
> 
> $ github-linguist Zelta/ --breakdown
> 
> 55.67%  8895       Inno Setup
> 
> 44.33%  7082       C#
> 
> 
> 
> Inno Setup:
> 
> Zelta-Installer.iss   <--- From above we can see this file alone is 8895 bytes vs 7082 for all your C#
> 
> 
> 
> C#:
> 
> Zelta/Form1.cs
> 
> Zelta/Program.cs
> 
> Zelta/Properties/AssemblyInfo.cs
> 
> Zelta/Settings.cs
> 
> Zelta/obj/Debug/.NETFramework,Version=v4.7.2.AssemblyAttributes.cs
> 
> 
> 
> $
> 
> ```
> 
> 
> 
> As you can see, it has only detected the one very large Inno Setup file and your C# files.
> 
> 
> 
> Why? Most of your files, like the C++ files you mentioned, are supplied as part of NuGet packages which are third-party packages which Linguist considered as vendored third-party code and thus [ignores](https://github.com/github/linguist/blob/97bc889ce840208652bf09b45f3b7859de43fe8e/lib/linguist/vendor.yml#L318-L319) (most people don't want third-party code counting towards their repo stats).
> 
> 
> 
> If you want these to count, you'll need to implement an [override](https://github.com/github/linguist/blob/master/docs/overrides.md) to tell Linguist to not consider these files as vendored third-party code.
> 
> 
> 
> The rest of your repo appears to be mostly "data" files because they are either binaries or are not UTF-8 encoded files, or text files that are considered "prose" (eg `.txt`) or "data" (eg `.xml`) which Linguist doesn't count towards the repo stats by default either. Once again, you'll need to implement an override if you want them to be counted.

"I needed this "
